src/huggingface_hub/hf_api.py

```diff
@@ -1474,11 +1503,26 @@ def list_models(
         path = f"{self.endpoint}/api/models"
         headers = self._build_hf_headers(token=token)
         params = {}
+
         if filter is not None:
             if isinstance(filter, ModelFilter):
                 params = self._unpack_model_filter(filter)
             else:
-                params.update({"filter": filter})
+                params = self._build_model_filter(
```

Comment:
I think it's best to have all the `_build_model_filter` inside `list_models`. Once `ModelFilter` are completely removed (and therefore `_unpack_model_filter`), we will have everything in one place which will be more consistent that the current code. Same way as you handled `pipeline_tag` below